pan3d/explorers/slice_explorer.py

```diff
@@ -0,0 +1,441 @@
+        self.outline_actor.SetMapper(self.outline_mapper)
+        self.outline_actor.GetProperty().SetColor(0.5, 0.5, 0.5)
+
+        self.data_actor          = vtk.vtkActor()        
+        self.data_mapper         = vtk.vtkDataSetMapper()
+        self.data_mapper.SetInputData(builder.mesh)
+        self.data_mapper.SetScalarRange(float(var_range[0]), float(var_range[1]))
+        self.data_actor.SetMapper(self.data_mapper)
+        self.data_actor.GetProperty().SetOpacity(0.1)
+
+        self.sbar_actor = vtk.vtkScalarBarActor()
+        self.sbar_actor.SetLookupTable(self.slice_mapper.GetLookupTable())
+        self.sbar_actor.SetMaximumHeightInPixels(600)
+        self.sbar_actor.SetMaximumWidthInPixels(150)
+        self.sbar_actor.SetTitleRatio(0.2) 
+        lprop : vtk.vtkTextProperty = self.sbar_actor.GetLabelTextProperty()
+        lprop.SetColor(0.5, 0.5, 0.5)
+        tprop : vtk.vtkTextProperty = self.sbar_actor.GetTitleTextProperty()
+        tprop.SetColor(0.5, 0.5, 0.5)
+
+        self.renderer.SetBackground(1., 1., 1.)
+        self.render_window.OffScreenRenderingOn()
+        self.render_window.AddRenderer(self.renderer)
+        self.interactor.SetRenderWindow(self.render_window)
+        self.interactor.GetInteractorStyle().SetCurrentStyleToTrackballCamera()
+
+        self.renderer.AddActor(self.outline_actor)
+        self.renderer.AddActor(self.slice_actor)
+        self.renderer.AddActor(self.sbar_actor)
+
+        self.renderer.ResetCamera()
+
+
+    @property
+    def ctrl(self):
+        return self.server.controller
+        
+    @property
+    def state(self):
+        return self.server.state
+
+    @property
+    def t_slice(self):
+        datavar = self.state.data_var
+        m       = self.state.time_active
+        mesh    = self.t_cache.get((m, datavar))
+        if mesh is None:
+            criteria = {}
+            criteria[self.builder.t] = m
+            da = self.builder.dataset[datavar].sel(**criteria)
+            da.load()
+            mesh = da.pyvista.mesh(x=self.builder.x, y=self.builder.y, z=self.builder.z)
+            self.t_cache[(m, datavar)] = mesh
+        return mesh
+ 
+    @property
+    def slice_dimension(self):
+        return self.state.slice_dim
+
+    @slice_dimension.setter
+    def slice_dimension(self, dim : str) -> None:
+        print(f"Setting slice dimension to {dim}")
+        self.state.slice_dim = dim
+        self.UpdateSlicer(dim)
+        self.on_data_change()
+
+    @property
+    def slice_value(self):
+        return self.state.dimval
+
+    @slice_value.setter
+    def slice_value(self, value : float) -> None:
+        print(f"Updating slice value tp {value}")
+        self.state.dimval = value
+        self.on_data_change()
+
+    @property
+    def view_mode(self):
+        return self.state.view_mode
+    
+    @view_mode.setter
+    def view_mode(self, mode):
+        self.state.view_mode = mode
```

Comment:
I would nest all the setters content inside `with self.state:` to allow out jupyter usage

Reply:
@ayenpure still not fixed

Reply:
Do you mean just like 
```
  @view_mode.setter
    def view_mode(self, mode):
      with self.state:
        self.state.view_mode = mode
        self.on_view_mode_change(mode)
```

Reply:
yes